src/ore/src/cast.rs

```diff
@@ -60,10 +60,12 @@ macro_rules! cast_from {
                 }
             }
 
-            /// Casts [`$from`] to [`$to`].
-            ///
-            /// This is equivalent to the [`crate::cast::CastFrom`] implementation but is
-            /// available as a `const fn`.
+            #[doc = concat!(
+                "Casts [`", stringify!($from), "`] to [`", stringify!($to), "`].\n",
+                "\n",
+                "This is equivalent to the [`crate::cast::CastFrom`] implementation but is\n",
+                "available as a `const fn`."
+            )]
```

Comment:
I think this isn't needed. We should just change the original docstring to say "Casts `from` to `to`." (i.e. drop the brackets and dollar signs.
```suggestion
            /// Casts `from` to `to`.
            ///
            /// This is equivalent to the [`crate::cast::CastFrom`] implementation but is
            /// available as a `const fn`.
```